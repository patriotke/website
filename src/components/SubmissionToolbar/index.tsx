/**
 *
 * SubmissionToolbar
 *
 */

'use client';

import TopicSelector from 'components/TopicSelector';
import { ITopicOption } from 'components/TopicSelector/ITopicSelectorProps';

type ISubmissionToolbarProps = {
  topics: ITopicOption[];
  onChange?: (field: string, value: string) => void;
};

function SubmissionToolbar({ topics, onChange }: ISubmissionToolbarProps) {
  const handleChange = (field: string) => (value: string) =>
    onChange?.(field, value);

  return (
    <div className="">
      <div className="py-2">
        <TopicSelector
          options={topics}
          placeholder="Select submission topic"
          label="Topic"
          onChange={handleChange('topic')}
        />
      </div>
    </div>
  );
}

export default SubmissionToolbar;
