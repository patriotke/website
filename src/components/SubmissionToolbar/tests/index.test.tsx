/* eslint-disable import/no-extraneous-dependencies */
import '@testing-library/jest-dom';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { render, screen } from '@testing-library/react';
import SubmissionToolbar from '../index';

describe('SubmissionToolbar', () => {
  it('renders SubmissionToolbar', () => {
    render(<SubmissionToolbar />);

    const { container } = render(<SubmissionToolbar />);
    expect(container).toMatchSnapshot();
  });
});
