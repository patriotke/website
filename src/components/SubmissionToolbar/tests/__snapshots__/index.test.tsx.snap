// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionToolbar renders SubmissionToolbar 1`] = `
<div>
  <div>
    SubmissionToolbar
  </div>
</div>
`;
